sql/src/test/java/io/crate/integrationtests/ShardingUpsertIntegrationTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.integrationtests;
+
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.junit.annotations.TestLogging;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 1)
+public class ShardingUpsertIntegrationTest extends SQLTransportIntegrationTest {
+
+    private String copyFilePath = Paths.get(getClass().getResource("/essetup/data/best_practice").toURI()).toUri().toString();
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    public ShardingUpsertIntegrationTest() throws URISyntaxException {
+    }
+
+    @Test
+    @TestLogging(value = "_root:DEBUG")
+    public void testCopyFromWithLimitedBulkSize() throws Exception {
```

Comment:
how can this test result in a retry, if the thread pool size is not adjusted? afaik no retry will ever happen currently